src/Test/Html/Events.elm

```diff
@@ -0,0 +1,60 @@
+module Test.Html.Events
+    exposing
+        ( msgFor
+        )
+
+{-|
+
+@docs msgFor
+-}
+
+import ElmHtml.InternalTypes exposing (ElmHtml(NodeEntry))
+import Json.Decode exposing (decodeString, decodeValue, field, string)
+import Native.HtmlAsJson
+import Test.Html.Query as Query
+import Test.Html.Query.Internal as QueryInternal
+
+
+getEventDecoder : String -> Json.Decode.Value -> Maybe (Json.Decode.Decoder msg)
+getEventDecoder =
+    Native.HtmlAsJson.getEventDecoder
+
+
+{-| Gets a Msg produced by a node when an event is triggered.
+
+    import Html
+    import Html.Events exposing (onInput)
+    import Test.Html.Query as Query
+    import Test.Html.Events as Events
+    import Test exposing (test)
+
+    type Msg
+        = Change String
+
+    test "Input produces expected Msg" <|
+        \() ->
+            Html.input [ onInput Change ] [ ]
+                |> Query.fromHtml
+                |> Events.msgFor "input" "{\"target\": {\"value\": \"cats\"}}"
```

Comment:
You could already use `Json.Encode` to generate it that way if you were so inclined...but I think it's a nice default choice for a test. If you mess it up, you'll presumably find out because the test will fail. 😄